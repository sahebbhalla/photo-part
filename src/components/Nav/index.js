import React from "react";


const Nav = (props) =>{
    const {
        categories=[],
        SetCurrentCategory,
        currentCategory
    } =props;

    return(
   <header className="flex-row px-1" style={{justifyContent:"space-between"}}>
        <h2 className="flex">
            <a data-testid="link" href="/">
                <span role ="img" aria-label="camera">{" "}ðŸ“¸</span>{" "}Oh snap    
            </a>
        </h2>
        <nav>
            <ul className="flex-row">
                <li className="mx-2">
                    <a data-testid="about" href="#about">About me</a>
                </li>
                <li>
                    <span>Contact</span>
                </li>
                {categories.map((category)=>(
                    <li className={`mx-1 ${currentCategory.name === category.name && 'navActive'}`} 
                    key={category.name}>
                        <span onClick={()=>SetCurrentCategory(category)}>{category.name}</span>
                    </li>
                ))}
            </ul>
        </nav>
   </header>
    )
}

export default Nav