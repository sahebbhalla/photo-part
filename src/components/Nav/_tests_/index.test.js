import React from 'react';
import{cleanup,render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';

afterEach(cleanup);

describe("Nav Component ", ()=>{
    test("renders",()=>{
        render(<Nav />);
    })
    test('matches Snampshot',()=>{
        const {asFragment} = render(<Nav />)
        expect(asFragment()).toMatchSnapshot()
    })
})
describe('emoji is visible', () => {
    test('inserts emoji into the h2', () => {
        const { getByLabelText } = render(<Nav />);
        expect(getByLabelText('camera')).toHaveTextContent('ðŸ“¸');
    
    })
  })

describe('Link Visibility',()=>{
    test("A href",()=>{
        const {getByTestId}= render(<Nav />);
        expect(getByTestId('link')).toHaveTextContent('Oh snap');
        expect(getByTestId('about')).toHaveTextContent('About');
    })
})