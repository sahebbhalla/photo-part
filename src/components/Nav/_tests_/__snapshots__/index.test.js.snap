// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component  matches Snampshot 1`] = `
<DocumentFragment>
  <header
    class="flex-row"
    style="justify-content: space-between;"
  >
    <h2
      class="flex"
    >
      <a
        data-testid="link"
        href="/"
      >
        <span
          aria-label="camera"
          role="img"
        >
          ðŸ“¸
        </span>
        Oh snap
      </a>
    </h2>
    <nav
      class="flex"
    >
      <ul
        class="flex-row"
      >
        <li
          class="mx-2"
        >
          <a
            data-testid="about"
            href="#about"
          >
            About me
          </a>
        </li>
        <li>
          <span>
            Contact
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            commercial
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            portraits
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            food
          </span>
        </li>
        <li
          class="mx-1"
        >
          <span>
            landscape
          </span>
        </li>
      </ul>
    </nav>
  </header>
</DocumentFragment>
`;
